modules/internal/ChapelArray.chpl

```diff
@@ -2397,7 +2391,7 @@ module ChapelArray {
       compilerError("Cannot remove indices from this domain type");
   }
 
-  proc -(d: domain, i: index(d)) where isIrregularDom(d) {
+  proc -(in d: domain, i: index(d)) where isIrregularDom(d) {
```

Comment:
Following up on the chat on this matter, I think this make the behavior much more like the intention of these procs.

Ideally, I would still like these args to be `const ref` where they are copied explicitly within the proc body instead of relying on `in` semantics. But that's probably more a matter of style.

Reply:
I probably would've done that as well, though I agree it's primarily a style question.  Playing devil's advocate, this approach _may_ let us "steal" a dead actual, re-use it within the routine, and return it without doing a copy, which would be cool (but I'm not confident that the compiler would/could do that today).

Reply:
Oops, I missed that you guys already came to this conclusion outside this thread.